saraho1123/currency-exchange
# Feature/display bookmarked

## Is this a fix or a feature?
feature

## What is the change?
+ The main change was to display cards that have been bookmarked by the user
  - Also added code to remove/add the "Add to bookmarked" button relative to if that card is in the bookmarked array
+ Added a delete button
  - Added a delete button to each card so a card can be deleted from the 'all' view or from the 'bookmarked view'
+ Added NavLinks:
  - user can navigate back and forth from currency-cards and bookmarked-conversions 
  - user can navigate back to the form (get a new conversion) from either the currency-cards or bookmarked-conversions 

## What does it fix?
+ App state was behind in sending the data from the app to the ExchangeContainer. I fixed it by passing the data from the api through arguments/params

## Where should the reviewer start?
+ line 6 of App.js
+ line 1 of Bookmarked.js
+ line 1 of BookmarkedContainer.js
+ line 4 of CurrencyCard.js
+ line 6 of ExchangeContainer.js

## How should this be tested?
+ clone down
+ cd into project
+ run `npm install` in terminal
+ run `npm start` in terminal
  - Form should appear. It can be filled out and button clicked, and user taken to `/currency-cards`.
  - The data from the user inputs, api data and manipulated api data would render
  - This can be done multiple times by click 'Get A New Currency Exchange' at the top of the page
  - While viewing currency conversion cards, user can add card to 'bookmarked'
  - Click 'See Your Bookmarked Conversions' to see cards that have been 'bookmarked'
  -
NOTE: styling and organization of data is incomplete at this point.

## Relevant Tickets
+ #3 
+ #4 
+ #5 
+ #6 
+ #7
